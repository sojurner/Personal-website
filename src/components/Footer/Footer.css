.footer {
  display: flex;
  position: relative;
  justify-content: center;
  bottom: 0;
  width: 100%;
  grid-row: 3 / 4;
}

.weather-content {
  border: 4px solid rgb(57, 125, 184);
  height: 7rem;
  width: 66rem;
  position: absolute;
  bottom: 1.7rem;
  left: 50%;
  margin: 0 0 0 -33rem;
  background-color: rgb(1, 126, 126);
  z-index: -1;
}

.rope-2 {
  position: absolute;
  bottom: 4.1rem;
  height: 16rem;
  width: 3rem;
  left: 11rem;
  border: 7px solid rgb(57, 125, 184);
  border-top: 0;
  background-color: #0f3f5db8;
}

.high-low {
  display: flex;
  justify-content: center;
  margin: 0 0 1rem 0;
}

.double-lines::before {
  content: '';
  position: absolute;
  height: 3rem;
  border-bottom: 7px solid rgb(57, 125, 184);
  border-top: 7px solid rgb(57, 125, 184);
  background-color: #0f3f5db8;
  top: 2.9rem;
  left: 13.3rem;
  width: 18%;
  z-index: 1;
}

.weather-high,
.weather-low {
  margin: 1rem 2rem 0 2rem;
  font-size: 3.3rem;
}

.weather {
  font-family: antaro;
}

.weather-low {
  color: rgb(150, 206, 249);
}

.weather-high {
  color: rgb(210, 142, 142);
}

.weather-icon-left,
.weather-icon-right {
  height: 70px;
  margin: 1rem 2rem 1rem 0;
}

.weather-icon-right {
  margin: 1rem 0rem 1rem 2rem;
}

.weather-summary {
  font-size: 3rem;
  color: rgb(183, 216, 245);
  /* margin: 0 0 3rem 0; */
}

@font-face {
  font-family: antaro;
  src: url(../../Fonts/Antaro.ttf);
}

.weather-info {
  font-weight: 100;
  color: #666;
  background-image: linear-gradient(to right, #0f3f5d, rgb(110, 146, 174));
  padding: 0 20px;
  font-size: 20px;
  line-height: 40px;
  max-width: 90vw;
  width: max-content;
  position: relative;
  text-align: center;
  height: 10rem;

  filter: drop-shadow(0 6px 3px rgba(0, 0, 0, 0.1));
  border-right: 2px solid #0f3f5d;
  border-left: 2px solid rgb(110, 146, 174);
}

.weather-info:after,
.weather-info:before {
  content: '';
  position: absolute;
  display: block;
  bottom: 0;
  border: 50px solid #0f3f5d;
}

.weather-info:after {
  border: 50px solid rgb(110, 146, 174);
}

.weather-info:before {
  left: -50px;
  border-right-width: 10px;
  border-left-color: transparent;
  filter: drop-shadow(-6px 6px 3px rgba(0, 0, 0, 0.04));
}

.weather-info:after {
  right: -50px;
  border-left-width: 10px;
  border-right-color: transparent;
  filter: drop-shadow(6px 6px 3px rgba(0, 0, 0, 0.04));
}
