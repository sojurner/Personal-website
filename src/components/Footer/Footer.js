import React from 'react';

import './Footer.css';

export const Footer = ({ weather }) => {
  return (
    <footer className="footer">
      <img
        alt="Weather icon"
        className="weather-icon"
        src={require(`../../assets/Images/weather/${weather.icon}.png`)}
      />
      <div className="weather-info">
        <div className="high-low">
          <h1 className="weather weather-high">↑{weather.high}°F</h1>
          <h1 className="weather weather-low">↓{weather.low}°F</h1>
        </div>
        <p className="weather weather-summary">{weather.summary}</p>
      </div>
      <img
        className="weather-icon"
        alt="Weather icon"
        src={require(`../../assets/Images/weather/${weather.icon}.png`)}
      />
    </footer>
  );
};
