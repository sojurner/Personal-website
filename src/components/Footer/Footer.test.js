import React from 'react';
import { Footer } from './Footer';

describe('Footer', () => {
  let wrapper;
  let mockWeather;
  beforeEach(() => {
    mockWeather = {
      icon: 'rain',
      high: 65,
      low: 20,
      summary: 'Its gonna be Cold man'
    };

    wrapper = shallow(<Footer weather={mockWeather} />);
  });

  it('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
