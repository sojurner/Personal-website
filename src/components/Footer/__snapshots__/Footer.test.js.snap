// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    weather={
      Object {
        "high": 65,
        "icon": "rain",
        "low": 20,
        "summary": "Its gonna be Cold man",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="rope-2"
        />,
        <span
          className="double-lines"
        />,
        <span
          className="weather-content"
        />,
        <img
          alt="Weather icon"
          className="weather-icon-left"
          src="rain.png"
        />,
        <div
          className="weather-info"
        >
          <div
            className="high-low"
          >
            <h1
              className="weather weather-high"
            >
              ↑
              65
              °F
            </h1>
            <h1
              className="weather weather-low"
            >
              ↓
              20
              °F
            </h1>
          </div>
          <p
            className="weather weather-summary"
          >
            Its gonna be Cold man
          </p>
        </div>,
        <img
          alt="Weather icon"
          className="weather-icon-right"
          src="rain.png"
        />,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rope-2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "double-lines",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "weather-content",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Weather icon",
          "className": "weather-icon-left",
          "src": "rain.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="high-low"
            >
              <h1
                className="weather weather-high"
              >
                ↑
                65
                °F
              </h1>
              <h1
                className="weather weather-low"
              >
                ↓
                20
                °F
              </h1>
            </div>,
            <p
              className="weather weather-summary"
            >
              Its gonna be Cold man
            </p>,
          ],
          "className": "weather-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="weather weather-high"
                >
                  ↑
                  65
                  °F
                </h1>,
                <h1
                  className="weather weather-low"
                >
                  ↓
                  20
                  °F
                </h1>,
              ],
              "className": "high-low",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "↑",
                    65,
                    "°F",
                  ],
                  "className": "weather weather-high",
                },
                "ref": null,
                "rendered": Array [
                  "↑",
                  65,
                  "°F",
                ],
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "↓",
                    20,
                    "°F",
                  ],
                  "className": "weather weather-low",
                },
                "ref": null,
                "rendered": Array [
                  "↓",
                  20,
                  "°F",
                ],
                "type": "h1",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Its gonna be Cold man",
              "className": "weather weather-summary",
            },
            "ref": null,
            "rendered": "Its gonna be Cold man",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Weather icon",
          "className": "weather-icon-right",
          "src": "rain.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="rope-2"
          />,
          <span
            className="double-lines"
          />,
          <span
            className="weather-content"
          />,
          <img
            alt="Weather icon"
            className="weather-icon-left"
            src="rain.png"
          />,
          <div
            className="weather-info"
          >
            <div
              className="high-low"
            >
              <h1
                className="weather weather-high"
              >
                ↑
                65
                °F
              </h1>
              <h1
                className="weather weather-low"
              >
                ↓
                20
                °F
              </h1>
            </div>
            <p
              className="weather weather-summary"
            >
              Its gonna be Cold man
            </p>
          </div>,
          <img
            alt="Weather icon"
            className="weather-icon-right"
            src="rain.png"
          />,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rope-2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "double-lines",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "weather-content",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Weather icon",
            "className": "weather-icon-left",
            "src": "rain.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="high-low"
              >
                <h1
                  className="weather weather-high"
                >
                  ↑
                  65
                  °F
                </h1>
                <h1
                  className="weather weather-low"
                >
                  ↓
                  20
                  °F
                </h1>
              </div>,
              <p
                className="weather weather-summary"
              >
                Its gonna be Cold man
              </p>,
            ],
            "className": "weather-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="weather weather-high"
                  >
                    ↑
                    65
                    °F
                  </h1>,
                  <h1
                    className="weather weather-low"
                  >
                    ↓
                    20
                    °F
                  </h1>,
                ],
                "className": "high-low",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "↑",
                      65,
                      "°F",
                    ],
                    "className": "weather weather-high",
                  },
                  "ref": null,
                  "rendered": Array [
                    "↑",
                    65,
                    "°F",
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "↓",
                      20,
                      "°F",
                    ],
                    "className": "weather weather-low",
                  },
                  "ref": null,
                  "rendered": Array [
                    "↓",
                    20,
                    "°F",
                  ],
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Its gonna be Cold man",
                "className": "weather weather-summary",
              },
              "ref": null,
              "rendered": "Its gonna be Cold man",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Weather icon",
            "className": "weather-icon-right",
            "src": "rain.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
